### PR TITLE
Upgrade minimum required and latest Cassandra version to 3.0 and 4.1 respectively

### PR DESCRIPTION
**Pull request by krvikash**
## Description

Fixes https://github.com/trinodb/trino/issues/14404

I am increasing the cassandra version for the test environment from 2.2 to 3.0. Minimum supported version is 3.0, see [here](https://cassandra.apache.org/_/download.html) and increasing the latest cassandra version to 4.1.

Both versions use different sets of properties, so added a separate file for the 4.1 version.
 

## Non-technical explanation
NA

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
(X) Release notes are required, with the following suggested text:

```markdown
# Cassandra
* Upgrade minimum required Cassandra version to 3.0. ({issue}`14562`)
* Upgrade latest Cassandra version to 4.1. ({issue}`14562`)
```

### CODE REVIEW

**Review by findepi — APPROVED** [review]

File: plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java [hunk 1]
```diff
@@ -69,7 +69,7 @@
     public CassandraServer()
             throws Exception
     {
-        this("cassandra:2.2");
+        this("cassandra:3.0");
```
[the remark this answers is not in the document]
Comment (ebyhr) [reply plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java:72]: Also, could you update the version in TestCassandraLatestConnectorSmokeTest to 4.1?

**Review by ebyhr — APPROVED** [review]
> (X) This is not user-visible or docs only and no release notes are required.

It would be better to mention the change to remove support of something. Let me update the description.

File: plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (krvikash) [reply plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java:72]: Addressed!

[the remark this answers is not in the document]
Comment (krvikash) [reply plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java:72]: Sure, WIll create a new PR for this.

[the remark this answers is not in the document]
Comment (krvikash) [reply plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java:72]: https://github.com/trinodb/trino/pull/14564 for latest cassandra version

**Comment by ebyhr** [comment]
Could you confirm CI failure?

File: plugin/trino-cassandra/pom.xml [hunk 2]
```diff
@@ -15,6 +15,7 @@
     <properties>
         <air.main.basedir>${project.parent.basedir}</air.main.basedir>
         <dep.native-protocol.version>1.5.1</dep.native-protocol.version>
+        <air.test.parallel>instances</air.test.parallel>
```
Comment (ebyhr) [inline plugin/trino-cassandra/pom.xml:18]: Was this property required to fix CI failure?

File: plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraConnectorTest.java [hunk 3]
```diff
@@ -553,10 +553,8 @@ public void testInsertToTableWithHiddenId()
     @Override
     public void testCreateTableWithLongColumnName()
     {
-        // TODO: Find the maximum column name length in Cassandra and enable this test.
         assertThatThrownBy(super::testCreateTableWithLongColumnName)
-                .hasMessageMatching(".* Mutation of .* bytes is too large.*");
-        throw new SkipException("TODO");
+                .hasMessageMatching(".*Attempted serializing to buffer exceeded maximum of 65535 bytes:.*");
```
Comment (ebyhr) [inline plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraConnectorTest.java:557]: We shouldn't override this method anymore. It should be handled in `verifyColumnNameLengthFailurePermissible ` method.

File: plugin/trino-cassandra/src/test/resources/cu-cassandra-latest.yaml [hunk 4]
```diff
@@ -0,0 +1,1794 @@
+# Cassandra storage config YAML
```
Comment (ebyhr) [inline plugin/trino-cassandra/src/test/resources/cu-cassandra-latest.yaml:1]: Could you put only required configuration likes cassandra-dse.yaml?

File: plugin/trino-cassandra/pom.xml (hunk 2 above)
Comment (krvikash) [reply plugin/trino-cassandra/pom.xml:18]: Yes this property fixed the CI failure.

File: plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraConnectorTest.java (hunk 3 above)
Comment (krvikash) [reply plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraConnectorTest.java]: Addressed! Removed the override method.

File: plugin/trino-cassandra/src/test/resources/cu-cassandra-latest.yaml (hunk 4 above)
Comment (krvikash) [reply plugin/trino-cassandra/src/test/resources/cu-cassandra-latest.yaml]: Modified the file with the minimum configuration. Though I could not find any doc on the what are properties that are not required to start the Cassandra server.

With the current property set, Tests are passing.

File: plugin/trino-cassandra/src/test/resources/cu-cassandra-latest.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+commitlog_segment_size: 32MiB
+seed_provider:
+    - class_name: org.apache.cassandra.locator.SimpleSeedProvider
+      parameters:
+          - seeds: "127.0.0.1:7000"
+concurrent_reads: 2
+concurrent_writes: 2
+concurrent_counter_writes: 2
+concurrent_materialized_view_writes: 2
+commitlog_total_space: 8192MiB
```
Comment (findepi) [inline plugin/trino-cassandra/src/test/resources/cu-cassandra-latest.yaml:18]: 8 GB sounds like a lot.
where is it stored?

Comment (krvikash) [reply plugin/trino-cassandra/src/test/resources/cu-cassandra-latest.yaml]: `commitlog_total_space: Total space to use for commit logs on disk.`
`Default Value: 8192MiB`
https://cassandra.apache.org/doc/latest/cassandra/architecture/storage_engine.html

For the test, I think we can reduce it to 1 MB same similar to https://github.com/trinodb/trino/blob/master/plugin/trino-cassandra/src/test/resources/cu-cassandra.yaml#L226.

**Pull request merged**